libsolidity/parsing/Parser.cpp

```diff
@@ -899,6 +899,23 @@ ASTPointer<VariableDeclaration> Parser::parseVariableDeclaration(
 	{
 		nodeFactory.markEndPosition();
 		tie(identifier, nameLocation) = expectIdentifierWithLocation();
+		if (
+			*identifier == "transient" &&
```

Comment:
I thought we were to introduce a `Token::Transient` token and allow it's usage where location specifiers are allowed currently - and only then forbid it in the analysis stage for everything other than state variable declarations.

Reply:
I didn't want to introduce a `Token::Transient` because we haven't made it a keyword yet.
But maybe that would be easier.